File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set<int>st;
        for(auto x:nums)st.insert(x);
        int ans=0;
        for(int i=0;i<nums.size();i++)
        {
            if(st.count(nums[i]-1))
            {
                continue;
            }
            int cn=0;
            int k=nums[i];
            while(st.count(k))
            {
                cn++;
                k++;
            }
            ans=max(ans,cn);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/jump-game/submission-0.java
class Solution {
    public boolean canJump(int[] nums) {

        int maxJump = 0;
        int n = nums.length;

        for( int i=0; i<n; i++){

            if(maxJump <= i+nums[i]){
                maxJump = Math.max(maxJump,i+nums[i]);

                if(maxJump >= n-1){
                    return true;
                }
            }

            if(maxJump == i){
                return false;
            }

        }

        return true;
        
    }
}